From jrv@vanzandt.mv.com Mon Jan 10 11:03:47 2000
Date: Sun, 9 Jan 2000 21:45:54 -0500 (EST)
From: James R. Van Zandt <jrv@vanzandt.mv.com>
To: vincent@ldsol.com
Subject: Re: y2k bug in adjtimex


Vincent -

This should do it.

	       - Jim Van Zandt

--- adjtimex.c-orig	Sun Jan  9 21:48:46 2000
+++ adjtimex.c	Sun Jan  9 21:50:53 2000
@@ -42,6 +42,8 @@
 #define WTMP_PATH "/var/log/wtmp"
 #endif
 
+static unsigned long epoch = 1900; /* year corresponding to 0x00   */
+
  /* Here the information for CMOS clock adjustments is kept. */
 #define ADJPATH "/etc/adjtime"
 
@@ -568,6 +570,8 @@
       tm.tm_mon--;		/* DOS uses 1 base */
       tm.tm_wday -= 3;		/* DOS uses 3 - 9 for week days */
       tm.tm_isdst = -1;		/* don't know whether it's daylight */
+      if ((tm.tm_year += (epoch - 1900)) <= 69)
+	tm.tm_year += 100;
 
       if (universal)
 	cmos_time = mkgmtime(&tm);
@@ -1175,6 +1179,8 @@
       tm.tm_mon--;		/* DOS uses 1 base */
       tm.tm_wday -= 3;		/* DOS uses 3 - 9 for week days */
       tm.tm_isdst = -1;		/* don't know whether it's daylight */
+      if ((tm.tm_year += (epoch - 1900)) <= 69)
+	tm.tm_year += 100;
 #ifdef DEBUG
       printf (" mday=%d  mon=%d  wday=%d  year=%d\n",
 	      tm.tm_mday, tm.tm_mon, tm.tm_wday, tm.tm_year);

